### PR TITLE
fix: preserve `view_class` in decorated views

### PR DESCRIPTION
**Pull request by Will-Hoey**
## Description

Add a check to preserve the `view_class` of a decorated function. Will allow retrieving a CBV name from a resolved URL

## Related resources

Open issue: https://github.com/django-cms/django-cms/issues/7653

## Checklist

* [x] I have opened this pull request against ``develop``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [ ] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: cms/tests/test_apphooks.py [hunk 1]
```diff
@@ -307,7 +307,7 @@ def test_apphook_permissions_preserves_view_name(self):
         view_names = (
             ('sample-settings', 'sample_view'),
             ('sample-class-view', 'ClassView'),
-            ('sample-class-based-view', 'view' if not (DJANGO_3 or DJANGO_2_2) else 'ClassBasedView' ),
+            ('sample-class-based-view', 'ClassBasedView' ),
```
Comment (marksweb) [inline cms/tests/test_apphooks.py:310]: We still support older versions of django, so I feel like this still needs conditional logic.

File: cms/utils/decorators.py [hunk 2]
```diff
@@ -20,6 +20,8 @@ def inner(request, *args, **kwargs):
         return func(request, *args, **kwargs)
     inner.__module__ = func.__module__
     inner.__doc__ = func.__doc__
+    if hasattr(func, "view_class"):
```
Comment (fsbraun) [inline cms/utils/decorators.py:23]: @Will-Hoey I'd prefer a conditional based on the Django version to keep the semantics and be able to remove the conditional completely once Django 3.2 is ancient history.

File: cms/tests/test_apphooks.py (hunk 1 above)
Comment (fsbraun) [reply cms/tests/test_apphooks.py:310]: @marksweb I think the PR is trying to achieve not having to have the Django version conditionals in custom code (or tests) but to always forward the view class name as presented by Django when using CMS decorators independently of the underlying Django version. @Will-Hoey Am I correct here?

File: cms/utils/decorators.py (hunk 2 above)
Comment (fsbraun) [inline cms/utils/decorators.py:23]: @Will-Hoey I retract my earlier comment on this line. Sorry.

File: cms/tests/test_apphooks.py (hunk 1 above)
Comment (Will-Hoey) [reply cms/tests/test_apphooks.py:310]: @fsbraun That is correct yes!

**Review by fsbraun — APPROVED** [review]
@Will-Hoey Great work! Thank you!

**Pull request merged**